include/seqan3/alignment/multiple/detail/align_multiple_seqan2_adaptation.hpp

```diff
@@ -0,0 +1,285 @@
+
+        size_t const alignment_length = concatenated_alignment_string.size() / output.size();
+
+        auto iter = output.begin();
+        for (auto && gapped_seq : concatenated_alignment_string | seqan3::views::char_to<gapped_alphabet_type>
+                                                                | seqan3::views::chunk(alignment_length))
+        {
+            iter->reserve(alignment_length);
+            std::ranges::copy(gapped_seq, std::cpp20::back_inserter(*iter++));
+        }
+
+        return output;
+    }
+
+private:
+    /*!\brief Predicate that represents whether `candidate_t` is a type that is not allowed as MSA configuration.
+     * \tparam candidate_t The type to test.
+     *
+     * \details
+     * The predicate's value is set to: NOT (band OR gap OR scoring).
+     */
+    template <typename candidate_t>
+    struct is_invalid_msa_config :
+        std::negation<std::disjunction<std::is_same<candidate_t, seqan3::align_cfg::band_fixed_size>,
+                                       is_type_specialisation_of<candidate_t, seqan3::align_cfg::gap>,
+                                       is_type_specialisation_of<candidate_t, seqan3::align_cfg::scoring_scheme>>>{};
+
+    /*!\brief Validate the given MSA configuration.
+     * \tparam configs_t The specified configuration elements.
+     */
+    template <typename ... configs_t>
+    void validate_configuration(seqan3::configuration<configs_t...> const &)
+    {
+        static_assert(seqan3::pack_traits::find_if<is_invalid_msa_config, configs_t...> == -1,
+                      "The given MSA configuration is not valid.");
+    }
+
+    /*!\brief Create the SeqAn2 scoring scheme based on the given SeqAn3 configuration.
+     * \tparam seqan3_configuration_t The type of the configuration object.
+     * \param config The configuration that contains scoring parameters.
+     * \return A SeqAn2 *MsaOptions* object with initialised scores.
+     *
+     * \details
+     * The scoring scheme matrices are copied from SeqAn3 into the corresponding SeqAn2 type.
+     * Note that the alphabet type can be different to that of the input sequences
+     * (seqan3::detail::align_multiple_seqan2_adaptation::alphabet_type), e.g., the SeqAn3 nucleotide scoring scheme is
+     * defined for dna16 but works for most other nucleotide alphabets too. The same holds for the SeqAn2 equivalent.
+     */
+    template <typename seqan3_configuration_t>
+    //!\cond
+        requires (seqan3_configuration_t::template exists<seqan3::align_cfg::scoring_scheme>())
+    //!\endcond
+    auto initialise_scoring_scheme(seqan3_configuration_t const & config)
+    {
+        auto scoring_scheme = get<seqan3::align_cfg::scoring_scheme>(config).value;
+
+        // A SeqAn3 scoring scheme might be of alphabet type dna15 although the aligned alphabet is dna4
+        // SeqAn3 alphabet type used:
+        using scoring_scheme_alphabet_type = typename decltype(scoring_scheme)::alphabet_type;
```

Comment:
I will add the check.

The thing is that I want to iterate over every pair of characters in the scoring scheme and set the score in the seqan2 scoring matrix. If I use the sequence alphabet (e.g. dna4) I will be missing some score (scoring scheme is dna15). I guess this won't make a problem now as we currently don't allow sequences of different types but it might in the future.